float32 pos_1
float32 pos_2
float32 pos_3
